src/lib/programmaticTransaction.ts

```diff
@@ -154,9 +188,84 @@ const revertAllChanges = async (maxIds: MaxIdRecord[]): Promise<void> => {
   }
 };
 
-const captureSnapshot = async (): Promise<MaxIdRecord[]> => fetchMaxIds();
+const captureSnapshot = async (
+  includeDDL = false,
+): Promise<MaxIdRecord[]> => fetchMaxIds(includeDDL);
 const rollbackToSnapshot = async (maxIds: MaxIdRecord[]): Promise<void> => revertAllChanges(maxIds);
 
+// This method of validating the transaction has not modified data is needed as the following
+// command fails on any creation of temp tables:
+// `SET TRANSACTION READ ONLY;`
+const hasModifiedData = async (snapShot, transactionId) => {
+  if (!transactionId) {
+    throw new Error('Transaction ID not found');
```

Comment:
I would assume this would trigger rollback.

Reply:
yes it does